web/src/pages/api/user/get.ts

```diff
@@ -9,53 +9,63 @@ export default async function get(req: NextApiRequest, res: NextApiResponse) {
     return res.status(401).json({ error: "You are not logged in" });
   }
 
-  const { userId, searchUserId } = req.query;
+  const { userId, searchUserId, userName } = req.query;
 
-  if (!searchUserId) {
-    return res.status(400).json("No searchUserId provided");
+  if (!searchUserId && !userName) {
+    return res.status(400).json({ error: "No user id or name provided" });
   }
 
-  if (Array.isArray(searchUserId)) {
-    return res.status(400).json({ error: "searchUserId cannot be an array" });
+  if (Array.isArray(searchUserId) || Array.isArray(userName)) {
+    return res
+      .status(400)
+      .json({ error: "The query params cannot be an array" });
   }
 
-  prisma
-    .$transaction([
-      prisma.user.findUnique({
-        where: {
-          id: searchUserId,
-        },
-        select: {
-          id: true,
-          name: true,
-          email: searchUserId === userId, // Only select this if the user is getting their own data
-          emailVerified: searchUserId === userId, // Only select this if the user is getting their own data
-          createdAt: true,
-          updatedAt: true,
-          image: true,
-        },
-      }),
-      prisma.collection.count({
-        where: {
-          userId: searchUserId,
-        },
-      }),
-      prisma.post.count({
-        where: {
-          authorId: searchUserId,
-        },
-      }),
-    ])
-    .then(([user, collectionCount, postCount]) => {
+  prisma.user
+    .findUnique({
+      where: {
+        id: searchUserId || undefined,
+        name: userName || undefined,
```

Comment:
If they provide both id and username and they aren't for the same user, what happens here?

Reply:
nothing would be sent back

Reply:
Right but is that the intended behavior? Are we trying to allow the caller to provide either `id` or `name`, or is providing both a legitimate option?

Reply:
ideally it would either be one or the other

Reply:
Ok then I think just a quick validation to ensure both aren't provided and then send back a 400 if both are